lib/awful/layout/init.lua.in

```diff
@@ -109,7 +109,15 @@ function layout.arrange(screen)
     p.geometry = capi.screen[screen].geometry
     p.clients = client.tiled(screen)
     p.screen = screen
-    layout.get(screen).arrange(p)
+
+    -- Apply layout geometry
+    local gs = layout.get(screen).arrange(p)
```

Comment:
Sure thing. Out of curiosity, what is the race condition? Is it that the client might be GCed before the geometry is set?

Also, this is my first time using Lua so I'll need to do some more reading on how weak tables work.